sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+twice at two different virtual addresses, one for mapping as code and
+the other for mapping as data.
+
+Shared objects are always built with a base virtual address of 0,
+while executables with an ELF ``e_type`` of ``ET_EXEC`` are built with
+a fixed virtual address. The underlying operating system maps shared
+objects in at different virtual addresses, utilizing the fact that the
+code and data addressing within such a shared object is position
+independent. The position independence relies on the principle that
+the distance between any two points within the image is identical at
+static link time and at run time.
+
+Program segments for executables built with a fixed virtual address
+have to be mapped identically to the virtual addresses used to create
+the executable file.
+
+Position independent executables are mapped in the same way as shared
+objects. They have an ELF ``e_type`` of ``ET_DYN``.
+
+Dynamic Section
+---------------
+
+The dynamic section entries allow information to be passed between the
+static linker and the dynamic linker. There are a number of entries
+that help locate various dynamic relocation tables and other aspects
+that are architecture independent.
+
+The AArch64 specific dynamic tags are defined in AAELF64_ Dynamic
+Section.
+
+Global Offset Table (GOT)
+-------------------------
+
+Position independent code cannot, in general, contain absolute (fixed)
+virtual addresses. Global offset tables hold absolute addresses in
+private data, thus making the addresses available without compromising
+the position independence and shareability of a program's text
+segment. A program references its global offset table using
+position independent addressing and extracts the absolute values from
+it, thereby redirecting position independent references to their
+actual locations.
+
+The global offset table (GOT) is created by the static linker in
+response to GOT generating relocations. See AAELF64_ Relocation
+operations for more information.
+
+AArch64 splits the global offset table (GOT) into two sections:
+
+ * ``.got.plt`` for code addresses accessed only from the Procedure Linkage
+   Table (PLT).
+
+ * ``.got`` all other addresses and offsets.
+
+Both the ``.got`` and ``.got.plt`` sections are aligned to a minimum
+of 8 bytes. They both contain 8-byte sized entries.
+
+If a program requires direct access to the absolute address of a
+symbol, that symbol will have a GOT entry. Because the executable and
+shared objects have their own independent GOTs, a symbol's address may
+appear in several tables. The dynamic linker processes all of the
+relocations in the ``.got`` section before giving control to any code
+in the process image, thus ensuring the absolute addresses are
+available during execution.
+
+By convention the ``.got`` section's first entry (number zero) is
+reserved to hold the address of the dynamic structure, referenced with
+the linker defined symbol ``_DYNAMIC``. Some dynamic linker's use this
```

Comment:
I think it isn't required now, but it would be for earlier versions of glibc. I wanted to find some words to express that, but not sure if I've succeeded.

Reply:
glibc dropped the `.got[0] == _DYNAMIC` requirement as a resolution to https://sourceware.org/bugzilla/show_bug.cgi?id=28203